### PR TITLE
Create python-app.yml

### PR DESCRIPTION
**Pull request by jafarsaad**

### CODE REVIEW

**Comment by l-lumin** [comment]
hi, what are the benefits of this workflow, considering we already have a `pre-commit.yml` and `tests.yml` workflow in place?

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
```
Comment (hamirmahal) [inline .github/workflows/python-app.yml:23]:
```suggestion
      uses: actions/setup-python@v5
```

`actions/setup-python@v3` will likely generate a warning about a deprecated Node.js version.

`v5` is the latest major version of https://github.com/actions/setup-python at the time of this writing.

**Review by hamirmahal — CHANGES_REQUESTED** [review]

**Comment by gagb** [comment]
@jafarsaad please see @l-lumin's comment

**Comment by gagb** [comment]
Appreciate the PR but it's redundant with GitHub Actions already in place. Please reopen if you don't agree. We appreciate your contributions.

**Pull request closed without merging**